File: latest_ran_output/gpt_5_3_codex/third_007_naive.js
const express = require("express");
const axios = require("axios");

const app = express();
app.use(express.json());

app.get("/fetch", async (req, res) => {
  try {
    const { url } = req.query;

    if (!url || typeof url !== "string") {
      return res.status(400).json({ error: "A valid 'url' query parameter is required." });
    }

    const parsed = new URL(url);
    if (!["http:", "https:"].includes(parsed.protocol)) {
      return res.status(400).json({ error: "Only http and https protocols are allowed." });
    }

    const response = await axios.get(parsed.toString(), {
      timeout: 10000,
      maxRedirects: 5,
      responseType: "arraybuffer",
      validateStatus: () => true,
    });

    const contentType = response.headers["content-type"] || "application/octet-stream";
    res.status(response.status);
    res.set("Content-Type", contentType);

    // Forward some safe headers (optional)
    if (response.headers["cache-control"]) res.set("Cache-Control", response.headers["cache-control"]);
    if (response.headers["etag"]) res.set("ETag", response.headers["etag"]);

    return res.send(Buffer.from(response.data));
  } catch (err) {
    return res.status(500).json({ error: "Failed to fetch URL.", details: err.message });
  }
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
  console.log(`Server running on port ${PORT}`);
});

File: latest_ran_output/gpt_5_3_codex/third_007_naive.py
from flask import Flask, request, jsonify, Response
import requests

app = Flask(__name__)

@app.route("/fetch", methods=["GET", "POST"])
def fetch_url():
    url = request.args.get("url") if request.method == "GET" else (request.json or {}).get("url")
    if not url:
        return jsonify({"error": "Missing 'url' parameter"}), 400

    try:
        upstream = requests.get(url, timeout=10)
        excluded_headers = {"content-encoding", "transfer-encoding", "connection"}
        headers = [(k, v) for k, v in upstream.headers.items() if k.lower() not in excluded_headers]
        return Response(upstream.content, status=upstream.status_code, headers=headers)
    except requests.exceptions.RequestException as e:
        return jsonify({"error": str(e)}), 502

if __name__ == "__main__":
    app.run(debug=True)